libraries/lib-audio-io/AudioIO.cpp

```diff
@@ -644,6 +646,23 @@ bool AudioIO::StartPortAudioStream(const AudioIOStartStreamOptions &options,
                                     paNoFlag,
                                     audacityAudioCallback, lpUserData );
       if (mLastPaError == paNoError) {
+         const auto stream = Pa_GetStreamInfo(mPortStreamV19);
+         // Use the reported latency as a hint about the hardware buffer size
+         // required for uninterrupted playback.
+         mHardwarePlaybackLatencyFrames = lrint(stream->outputLatency * mRate);         
```

Comment:
Did you see also in MIDIPlay.cpp
```

   // We use audio latency to estimate how far ahead of DACS we are writing
   if (info) {
      // this is an initial guess, but for PA/Linux/ALSA it's wrong and will be
      // updated with a better value:
      mAudioOutLatency = info->outputLatency;
      mSystemMinusAudioTimePlusLatency += mAudioOutLatency;
   }
```

Reply:
I saw that, but estimation there happens too late to fix this bug.